### PR TITLE
fixing title and subtitles bug

### PR DESCRIPTION
**Pull request by silviafan**

### CODE REVIEW

**Review by wiheto — CHANGES_REQUESTED** [review]
You do not need to keep calling _add_subtitle_title functions so many times. 

all you need is

if profile['subtitles'] is None and profile['title'] == 'auto':
    profile['subtitles'] = 'auto'
    profile['title'] = None

And then call _add_subplot_title and _add_title

I do not understand why everything else is needed.

I do not understand the rest of the difficulty and why there are so many condition

File: netplotbrain/plot.py [hunk 1]
```diff
@@ -223,8 +220,41 @@ def plot(nodes=None, fig: Optional[plt.Figure] = None, ax=None, view: str = 'L',
                 _plot_springlayout(ax, nodes=nodes, edges=edges, node_color=node_color, node_size=node_size,
                                    edge_color=edge_color, edge_weights=edge_weights, highlight_nodes=highlight_nodes, **profile)
             ax.view_init(azim=azim[fi], elev=elev[fi])
-
-            _add_subplot_title(ax, azim[fi], elev[fi], subtitle_frame, hemi_frame, viewtype[fi], **profile)
+
```
Comment (wiheto) [inline netplotbrain/plot.py:223]: This needs to be made more concise. 

I only see only option needed:

if there is no input for title or subtitle... then title is 'auto' and subtitles is None (or maybe it should be 'auto' too)

When then occurs, set title to None and subtitles to auto. Why are any other automatic scenario needed other than this.

File: netplotbrain/plot.py [hunk 2]
```diff
@@ -253,11 +264,11 @@ def plot(nodes=None, fig: Optional[plt.Figure] = None, ax=None, view: str = 'L',
             ax.axis('off')
             #ax = _add_size_legend(ax, nodes, node_size, node_scale)
             ax_out.append(ax)
-
+        
     # Title on top of the figure
-    if profile['title'] is not None:
+    if profile['title'] is not None:
```
Comment (wiheto) [inline netplotbrain/plot.py:269]: remove trailing white space

File: netplotbrain/plotting/plot_title.py [hunk 3]
```diff
@@ -6,20 +6,20 @@ def _add_subplot_title(ax, azim=None, elev=None, subtitle_frame='auto', hemisphe
     subtitle_fontsize = kwargs.get('subtitle_fontsize')
     subtitle_loc = kwargs.get('subtitle_loc')
     subtitle_weight = kwargs.get('subtitle_weight')
-    if subtitle_frame == 'auto':
+    if subtitle_frame=='auto':
```
Comment (wiheto) [inline netplotbrain/plotting/plot_title.py:9]: add space after between ==

File: netplotbrain/plot.py [hunk 4]
```diff
@@ -125,7 +125,17 @@ def plot(nodes=None, fig: Optional[plt.Figure] = None, ax=None, view: str = 'L',
             view, hemisphere = _get_presetviews(view)
     # Get number of non-legend rowsnon
     nrows, view, frames = _nrows_in_fig(view, frames)
-          
+    
+    if profile['subtitles'] == 'auto' and profile['title'] == 'auto':
```
[the remark this answers is not in the document]
Comment (wiheto) [reply netplotbrain/plot.py:129]: Otherwise looks really good

**Comment by wiheto** [comment]
Looks better!
Only some minor things left

**Comment by silviafan** [comment]
Great, all fixed! 👍

**Pull request merged**